### ISSUE TITLE
Allow gravity to be relative

### ISSUE THREAD

**Issue by dunklesToast** [imgproxy/imgproxy#1228]
Since we can use Crop with relative values (0-1) it'd be great to also make gravity support those.

Let's take this 200x200 Image as an example:
![test-crop3](https://github.com/imgproxy/imgproxy/assets/17279485/332d5831-377d-46af-b7e9-d5ff73086268)

Using these settings `crop:0.5:0.5/g:nowe:100:100` we're just getting the blue square because (as far as I understand it) imgproxy will use the coordinate (100|100) as a gravity point and then crops 50% of the whole image width to the right and 50% of the image height to the bottom resulting in the Points (100|100) and (200|200) as boundaries. 

It'd be great if we could also use relative values for the gravity, so that `crop:0.5:0.5/g:nowe:0.5:0.5` results in the same output. Any thoughts about this or is it already possible and I am just overseeing something?

**Comment by DarthSim** [comment]
Hey @dunklesToast!

I believe, the `fp` gravity type is what you are looking for

**Comment by dunklesToast** [comment]
Hey @DarthSim 

Thanks for the prompt reply - I actually fiddled around with `fp` but seems that i cannot use it due to this
> that define the coordinates of the center of the *resulting image*

Just to give some more context what I am trying to do:

We're using [react-image-crop](https://www.npmjs.com/package/react-image-crop) in our frontend to get a cropping area of images. This package outputs the selected crop in either pixels or percentages and uses the top left (northwest) as origin.
The output of the cropper can look something like this:

```json
{
    "unit": "%",
    "x": 33,
    "y": 0,
    "width": 66,
    "height": 50
}
```

The according selection would be this
<img width="213" alt="image" src="https://github.com/imgproxy/imgproxy/assets/17279485/7a82e25d-b010-408b-9548-3a619c29ffa2">

With the "normal" gravity options supporting relative values, I could easily do something like `crop:0.66:0.5/g:nowe:0.33:0`.
I've validated this approach by using absolute values as I know the proportions of this example image: Using `crop:0.66:0.50/g:nowe:66:0` (66px as the image is 200px wide and 33% of that is 66) results in the correctly cropped image
![public](https://github.com/imgproxy/imgproxy/assets/17279485/361fc507-1da7-4832-889d-eaf928f2c116)

Since `fp` uses the center of the already cropped image, I'd need to do some calculations based on the images width and the cropped center point to the calculate the correct `fp` values where I am not currently sure how to it - I'll play around with that a bit, but if you have any ideas or suggestions let me know. IMO having `gravity` support relative values as well would be the easiest as it allows to define the origin but I'll try using. `fp`.
